src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -15,21 +15,19 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Definitions
 
-(def jruby-puppet-env
-  "The environment variables that should be passed to the Puppet JRuby interpreters.
-  We don't want them to read any ruby environment variables, like $GEM_HOME or
-  $RUBY_LIB or anything like that, so pass it an empty environment map - except -
-  Puppet needs HOME and PATH for facter resolution, so leave those."
-  (select-keys (System/getenv) ["HOME" "PATH"]))
-
 (def ruby-code-dir
-  "The name of the directory containing the ruby code in this project.
-  This directory lives under src/ruby/"
-  "puppet-server-lib")
+  "The name of the directory containing the ruby code in this project."
+  "src/ruby/puppet-server-lib")
```

Comment:
@KevinCorcoran Yeah, understood that it's a relative path inside the jar, but the behavior is that it's added to the $LOAD_PATH (which is funky) and JRuby looks in JAR files by pre-pending each element of the LOAD_PATH to the value passed to "require" then peeking inside the JAR root.

So, even though it's used for a JAR file, it's added to the LOAD_PATH which affects both the behavior of filesystem and JAR resource loading.